### DISCUSSION TITLE
Does Tycho surefire support actionSets menus?

### DISCUSSION THREAD

**Discussion by Hello-dl** [eclipse-tycho/tycho#518] [Q&A] [answered]
We have an old application build on RCP 3.x and its menu was created using ActionSet; We created a separate Maven project in Eclipse JEE to use Tycho-Surefire-plugin to run cucumber test scripts for regression testing. However, Tycho-surefire did not successfully load its menu items when launching the application while the application can run successfully on its own. I would like to confirm here if Tycho surefire does not support actionSets menus?

**Comment by mickaelistria** [comment]
Tycho surefire just starts an Eclipse application and doesn't add much behavior content (apart of running the tests against the started application). The cause of your problem is not Tycho surefire, but maybe your test missing a dependency that make your actionSets working as expected. You can debug running tests with https://www.eclipse.org/tycho/sitedocs/tycho-surefire-plugin/test-mojo.html#debugPort

**Reply by Hello-dl** [reply]
Thanks for your suggestion, we are currently using Maven’s command line to implement debugging：
mvn clean verify -Xdebug -Xrunjdwp:transport=dt_socket,address=8000,server=y,suspend=y
I wonder if this method is different from Tycho Surefire’s parameter your suggested. 
Besides, this method can only debug the test script and cannot enter the breakpoint in the script of the application which are testing

**Comment by akurtakov** [comment, marked as answer]
Tycho itself has no notion of actionSet, command and etc. It all depends on the target platform which you build/run tests your application with. Please share smallest possible reproducer project that fails in that way so it's possible to see the full log and experiment. 
I would suggest moving to latest version of all dependencies as Eclipse Platform 3.x is not supported for ages but 3.x style RCPs are supposed to run just fine with latest Eclipse Platform 4.22 release.

**Reply by Hello-dl** [reply]
Thanks for your reply, currently, the application under test has been upgraded to Eclipse 4.18+java 11. It's a little bit difficult to share the whole project here, may I share the logs with you first? to see if it's help to confirm the issue more, please check the logs attached, one is log which the application started by itself, another one is log started by tycho surefire
[Started by application its own.log](https://github.com/eclipse/tycho/files/7837385/Started.by.application.its.own.log)
[Started by tycho.log](https://github.com/eclipse/tycho/files/7837386/Started.by.tycho.log)

**Reply by akurtakov** [reply]
So the target platform you run tests with differs from the way you run things. Unfortunately, the logs show the errors but we can't give you any hint how to fix it without being able to look at the project.
